src/target/adiv5.c

```diff
@@ -438,6 +438,9 @@ void adiv5_dp_init(ADIv5_DP_t *dp)
 		if (ap == NULL)
 			continue;
 
+		extern void kinetis_mdm_probe(ADIv5_AP_t *);
```

Comment:
I agree having this two way dependency isn't very nice, but I'm leaving it as it is for now.  We should clean this up and have drivers register themselves with their parents in future, but that is a significant refactor that is out of scope of this PR.